Guides/Trim.md

```diff
@@ -0,0 +1,116 @@
+# Trim
+
+[[Source](https://github.com/apple/swift-algorithms/blob/main/Sources/Algorithms/Trim.swift) |
+ [Tests](https://github.com/apple/swift-algorithms/blob/main/Tests/SwiftAlgorithmsTests/TrimTests.swift)]
+
+Returns a `SubSequence` formed by discarding all elements at the start and end of the Collection
```

Comment:
I think the prevailing style is to use code voice to refer to a `Thing` or otherwise to refer to the thing being modeled in lowercase. A `Collection` is a protocol for collections; a `String` models a string.

```suggestion
Returns a `SubSequence` formed by discarding all elements at the start and end of the collection
```